lib/viewcomponents/io.jsx

```diff
@@ -0,0 +1,78 @@
+var React = require('react');
+var mouseDownDrag = require('../utilities/mouseDownDrag');
+var LabelPosition = require('../enum/ioLabelPosition');
+
+var ioRadius = 4;
+
+var IO = React.createClass({
+  propTypes: {
+    isInput: React.PropTypes.bool.isRequired,
+
+    label: React.PropTypes.string,
+
+    x: React.PropTypes.number.isRequired,
+
+    y: React.PropTypes.number.isRequired,
+
+    size: React.PropTypes.number,
+
+    labelPosition: React.PropTypes.oneOf(Object.keys(LabelPosition).map(function(k) {
+      return LabelPosition[k];
+    }))
```

Comment:
I don't really understand what you're doing here. I get the oneOf and the map function, but what's up with Object.keys? I've never seen that.